### ISSUE TITLE
Form inputs display issues on Safari

### ISSUE THREAD

**Issue by lcgiry** [stephanrauh/ngx-extended-pdf-viewer#2453]
ngx-extended-pdf-viewer version : 18.1.14
Angular version : 15

Hello,

We are experiencing some issue with PDFs containing forms on Safari. When some inputs are filled, their value is not always displayed. When not displayed, resizing the window make the input values blink but they are still invisible when resizing is done.

In the two following screenshots, we used the same ngx-extended-pdf-viewer configuration. On is part of a Material Dialog that opens in full screen, and the malfunctioning one is part of a page.

You can find attached the PDF used for this tests.
Thanks !

```
    <ngx-extended-pdf-viewer
      [src]="'/assets/Mouvement n°4 - cerfa n°2759-SD.pdf'"
      [disableForms]="true" 
      [showToolbar]="false" [sidebarVisible]="false" [showUnverifiedSignatures]="true" [showBorders]="false" [ignoreKeyboard]="true" [backgroundColor]="undefined" [enableDragAndDrop]="false" [enablePrint]="false" theme="dark">
    </ngx-extended-pdf-viewer>
```
Nominal behavior
![Screenshot 2024-07-22 at 15 52 18](https://github.com/user-attachments/assets/309915f9-158d-470e-8ae9-047cb9de9c25)

Malfunctioning viewer
![Screenshot 2024-07-22 at 15 50 33](https://github.com/user-attachments/assets/c60a401a-8507-46ac-a54b-e19a23ac8265)

[Mouvement n°4 - cerfa n°2759-SD.pdf](https://github.com/user-attachments/files/16337243/Mouvement.n.4.-.cerfa.n.2759-SD.pdf)

**Comment by stephanrauh** [comment]
Strange. The forms are displayed correctly on other browsers?

**Comment by stephanrauh** [comment]
And how did you fill the input fields? Via Angular or via user input?

**Comment by lcgiry** [comment]
Yes, the input fields are displayed properly on every other browser we tested (Firefox, Chrome and Edge).
We filled the inputs using pdf-lib and tried to edit them directly in ngx-extended-pdf-viewer, it doesn't  seem to change anything.

**Comment by walweh** [comment]
I'm facing the same issue with form fields not displaying properly. Any updates or solutions?"

**Comment by stephanrauh** [comment]
Not yet. Can you give me any hint on how to reproduce the issue?

**Comment by walweh** [comment]
- The issue primarily occurs on the second attempt to open the PDF in ngx-extended-pdf-viewer.

- Forms within the PDF are not visible initially, but they can still be filled out.

-  When zooming in, the forms briefly appear but then disappear.

**Comment by walweh** [comment]
- The issue primarily occurs on the second attempt (for me)  to open the PDF in ngx-extended-pdf-viewer.
- Forms within the PDF are not visible initially, but they can still be filled out.
- When zooming in, the forms briefly appear but then disappear.

- When using the pageViewMode set to infinite-scroll, the PDF forms are initially visible upon opening the document.
- However, the issue arises when zooming in: the forms briefly appear but then disappear.
- This problem does not occur when zooming out; the forms remain visible during zoom-out operations.
after zooming in, the forms appear to go under the document content, becoming invisible but you can still fill them.


also testet it with the document of @lcgiry  

I hope it helps a little

**Comment by walweh** [comment]
![0fcd8966-828e-4242-8240-4d84e3e9f6ee](https://github.com/user-attachments/assets/d91b2d73-bc77-4f49-980a-76be05ec609a)
![2f8cc079-1c76-46a0-9a87-e941025e000c](https://github.com/user-attachments/assets/5189c43c-bb8e-4124-bee7-b643ff91052a)

**Comment by walweh** [comment]
I moved the layer a bit so you can see what I mean

**Comment by durgamalleswariyeruva** [comment]
any update on this issue?

**Comment by stephanrauh** [comment]
@lcgiry @walweh Sorry for responding late. I can't reproduce your issue with the current version of the viewer. Please update to ngx-extended-pdf-viewer 24.2 and tell me if the bug is gone. That's not unlikely: version 18.1 used pdf.js 3.9, and the current version uses pdf.js 5.3. The pdf.js team have solved quite a few bugs in the meantime.

If the bug is still there, or if you can't update to a new version, try adding this CSS to the style.(s)css:

```css
.ngx-extended-pdf-viewer {
  -webkit-transform: translateZ(0);
  transform: translateZ(0);
  will-change: transform;
}
```

This CSS activates GPU acceleration. I'm told Safari has many issues with form rendering, and they're usually gone once GPU acceleration is active.

If that doesn't work, try adding this:

```css
.ngx-extended-pdf-viewer input[type="text"] {
  position: relative;
  z-index: 1;
}
```

I don't think that makes any difference, but it's worth a try. Obviously, Safari is confused, maybe it needs some help.

Talking of which: I'd appreciate your feedback. I'll close the issue for the moment because I couldn't reproduce it, but I'll re-open it if necessary.
